### ISSUE TITLE
[BUG] & [FR] In write, the source field is not checked correctly

### ISSUE THREAD

**Issue by HarikalarKutusu** [common-voice/common-voice#4559]
**Describe the bug**
The source field on write page form is mandatory. If you leave it empty, it will complain.

On the other hand, you can just put a space there to overcome the issue as it is only checked against empty string.
It will be better to check at least after a trim operation.

**Related Feature Request**

There can be infinite sources, but one single source is common among languages: self-written sentences. You can fill in anything there to specify that these are your own words, many different formats, which cannot be analyzed from the dataset release files (My own words, I wrote it, etc. etc.).
If we can have a checkbox/button there to fill a standardized sentence (from translation files) it can make filling that field easy for most cases, also better to analyze.

PS: In my case, during validation, when it says "My own words" (like) and the phrase is not a common conversational one, I check search it in Google to check if it is a copy-paste from a copyrighted source.

**Comment by jgrant29** [comment]
To clarify, you're requesting a standardized way to mark a sentence as self-written without having to type it because not all sources are using the same citation for self-written?

**Comment by HarikalarKutusu** [comment]
Exactly. From the legal point, we already have a checkbox and an additional button to be clicked, so there is no need for a person to say that he/she is making the self produced/written sentence CC0/Public Domain.

As a suggestion: There can be a (localized) standard string which can be inserted when I button is clicked. This way we can extract these strings from the text corpus and check against Internet for example to see if they exist in copyrighted sources (even if the person says I wrote it, I see many technical sentences which look like taken from a book).

Free form fields are always problematic in this respect. Same happened for the accent field in the profile in the past, which became impossible to analyze.

Not the most important thing now, but anyway. The bug part is important thou...

**Comment by jgrant29** [comment]
Thank you for clarifying and sharing, @HarikalarKutusu!  We appreciate all of your support and feedback with the platform.  It's really great having a community person like you contributing and helping make the platform better for all. 

I will add this to our team discussion and circle back to you about progress with this.  Thanks again!

**Comment by moz-kathyreid** [comment]
Confirming that the bug around adding a space in the Citation field allows the sentence to be submitted still exists. 

Flagging as Data Quality label, because if sentences have been added without a valid citation then it makes it more difficult to establish the provenance of the sentence and ensure it's CC-0.

**Comment by HarikalarKutusu** [comment]
This turns out to be a simple bug caused by bad Fluent call. It seems to be left-over from transition to multiple translation files.

Existing form:
```
      <Localized id="sc-my-source" vars={{ source }}>
        <p className="source" />
      </Localized>
```

In the new order, this actually should be:
```
      <Localized id="sc-review-form-source" vars={{ sentenceSource: source }}>
        <p className="source" />
      </Localized>
```

This is defined in `review.ftl` as:
```
sc-review-form-source = Source: { $sentenceSource }
```

Old form was (from the left over German translation file):
```
sc-my-source = Quelle: { $source }
```
Sending a PR...

**Comment by moz-bozden** [comment]
@moz-kathyreid, passing an empty field in source field is actually fixed with this not-yet-reviewed PR from March 28 (by my other self):
https://github.com/common-voice/common-voice/pull/4850

**Comment by moz-kathyreid** [comment]
Nicely identified, @moz-bozden!

**Comment by HarikalarKutusu** [comment]
This has been fixed & merged.
